moonraker/components/wled.py

```diff
@@ -0,0 +1,445 @@
+        else:
+            # If not transmitting this time easiest just to send all data when
+            # next transmitting
+            self.send_full_chain_data = True
+
+class WLED:
+    def __init__(self: WLED, config: ConfigHelper) -> None:
+        try:
+            # root_logger = logging.getLogger()
+            # root_logger.setLevel(logging.DEBUG)
+
+            self.server = config.get_server()
+            prefix_sections = config.get_prefix_sections("wled")
+            logging.info(f"WLED component loading strips: {prefix_sections}")
+            color_orders = {
+                "RGB": ColorOrder.RGB,
+                "RGBW": ColorOrder.RGBW
+            }
+            self.strips = {}
+            for section in prefix_sections:
+                cfg = config[section]
+
+                name_parts = cfg.get_name().split(maxsplit=1)
+                if len(name_parts) != 2:
+                    raise cfg.error(
+                        f"Invalid Section Name: {cfg.get_name()}")
+                name: str = name_parts[1]
+
+                logging.info(f"WLED strip: {name}")
+
+                color_order_cfg: str = cfg.get("color_order", "RGB")
+                color_order = color_orders.get(color_order_cfg)
+                if color_order is None:
+                    raise config.error(
+                        f"Color order not supported: {color_order_cfg}")
+
+                self.strips[name] = Strip(name, color_order, cfg)
+
+            event_loop = self.server.get_event_loop()
+            event_loop.register_callback(
+                self._initalize_strips, list(self.strips.values()))
+
+            # Register two remote methods for GCODE
+            self.server.register_remote_method(
+                "set_wled_state", self.set_wled_state)
+            self.server.register_remote_method(
+                "set_wled", self.set_wled)
+
+            # As moonraker is about making things a web api, let's try it
+            # Yes, this is largely a cut-n-paste from power.py
+            self.server.register_endpoint(
+                "/machine/wled/strips", ["GET"],
+                self._handle_list_strips)
+            self.server.register_endpoint(
+                "/machine/wled/status", ["GET"],
+                self._handle_batch_wled_request)
+            self.server.register_endpoint(
+                "/machine/wled/on", ["POST"],
+                self._handle_batch_wled_request)
+            self.server.register_endpoint(
+                "/machine/wled/off", ["POST"],
+                self._handle_batch_wled_request)
+            self.server.register_endpoint(
+                "/machine/wled/strip", ["GET", "POST"],
+                self._handle_single_wled_request)
+
+        except Exception as e:
+            logging.exception(e)
+
+    async def _initalize_strips(self: WLED,
+                                initial_strips: List[Strip]
+                                ) -> None:
```

Comment:
This method should be renamed to `component_init()`.  It can't take an argument (outside of `self`) so you will need to directly reference `self.strips.values()` instead of `initial_strips`.

Moonraker will call the `component_init()` method after the event loop starts but before the http server starts.

Reply:
Thanks, working again now